GetStream/stream-chat-android
# [CAS-884] Add possibility to group notifications

https://stream-io.atlassian.net/browse/CAS-884

### 🎯 Goal

Add possibility to group notifications and dismiss all notifications related to the particular channel after notification click.

### 🛠 Implementation details

Added additional notification (group summary) which is displayed every time we receive a new notification and grouping is enabled in config. The notification is responsible for grouping messages from a particular channel and dismissing all notifications (from that group) when it's clicked.
NOTE: If you click on a single notification from the notification's group, other notifications that belong to this group won't be dismissed. Check video for further details

https://user-images.githubusercontent.com/17440581/118822549-8f251f00-b8b8-11eb-8630-2402894f3047.mp4


### 🧪 Testing

1. Send few messages from another device
2. Click on notification summary
3. All notifications related to particular channel should be removed

### ☑️ Checklist
